idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/lineMarker/LineMarkerSyntax.kt

```diff
@@ -39,32 +43,113 @@ interface LineMarkerSyntax {
     )
 
   /**
    * [addLineMarkerProvider] is a convenience extension, which registers the Leaf element of a composite PsiElement [A] e.g.: `KtClass`
    * and circumvents effort's to find the right PsiElement.
    * In addition, plugin developer's can compose sophisticated messages, as the whole scope of [A] can be exploited.
+   * ```kotlin:ank:playground
+   * import arrow.meta.Plugin
+   * import arrow.meta.ide.IdeMetaPlugin
+   * import arrow.meta.ide.resources.ArrowIcons
+   * import arrow.meta.invoke
+   * import com.intellij.psi.PsiElement
+   * import org.jetbrains.kotlin.idea.caches.resolve.resolveToDescriptorIfAny
+   * import org.jetbrains.kotlin.psi.KtNamedFunction
+   * import org.jetbrains.kotlin.renderer.ParameterNameRenderingPolicy
+   * import org.jetbrains.kotlin.utils.addToStdlib.safeAs
+   *
+   * val IdeMetaPlugin.descriptorLineMarker: Plugin
+   *   get() = "Render Descriptor Plugin" {
+   *     meta(
+   *       addLineMarkerProvider(
+   *         transform = { e: PsiElement -> e.safeAs() },
+   *         composite = KtNamedFunction::class.java,
+   *         icon = ArrowIcons.ICON2,
+   *         message = { f ->
+   *           HTML.withOptions { // check out DescriptorRenderer's companion for more options
+   *             unitReturnType = true // renders Unit Type
+   *             classifierNamePolicy = classifierNamePolicy() // define your own policies
+   *             parameterNameRenderingPolicy = ParameterNameRenderingPolicy.ONLY_NON_SYNTHESIZED
+   *           }.let { renderer ->
+   *             f.resolveToDescriptorIfAny()?.let(renderer::render) ?: "Unresolved Descriptor"
+   *           }
+   *         }
+   *       )
+   *     )
+   *   }
+   * ```
    * @param composite In Contrast, lineMarkers constructed without this parameter have a clearly constrained message.
+   * @param message you may use a [DescriptorRenderer] for rendering descriptors see [descriptorRender]
    */
   @Suppress("UNCHECKED_CAST")
   fun <A : PsiNameIdentifierOwner> IdeMetaPlugin.addLineMarkerProvider(
     icon: Icon,
     transform: (PsiElement) -> A?,
     composite: Class<A>,
-    message: (A) -> String = Noop.string1(),
-    placed: GutterIconRenderer.Alignment = GutterIconRenderer.Alignment.RIGHT
+    message: DescriptorRenderer.Companion.(A) -> String = Noop.string2(),
+    placed: GutterIconRenderer.Alignment = GutterIconRenderer.Alignment.RIGHT,
+    navigate: (event: MouseEvent, element: PsiElement) -> Unit = Noop.effect2,
+    clickAction: AnAction? = null
   ): ExtensionPhase =
     addLineMarkerProvider(
       { transform(it)?.identifyingElement },
       { identifier: PsiElement ->
         PsiTreeUtil.getParentOfType(identifier, composite)?.let { psi: A ->
-          lineMarkerInfo(icon, identifier, { message(psi) }, placed)
+          lineMarkerInfo(icon, identifier, { message(DescriptorRenderer.Companion, psi) }, placed, navigate, clickAction)
```

Comment:
`DescriptorRenderer` is from the kotlin plugin, but this method doesn't seem to be limited to Kotlin PSI.